devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -528,49 +528,45 @@ void prepare() throws Exception {
+                addProject(devModeContext, localProject);
+                pomFiles.add(localProject.getDir().resolve("pom.xml"));
+            } else {
+                localProject = LocalProject.loadWorkspace(outputDirectory.toPath());
+                for (LocalProject project : localProject.getSelfWithLocalDeps()) {
+                    if (project.getClassesDir() != null) {
+                        //if this project also contains Quarkus extensions we do no want to include these in the discovery
+                        //a bit of an edge case, but if you try and include a sample project with your extension you will
+                        //run into problems without this
+                        if (Files.exists(project.getClassesDir().resolve("META-INF/quarkus-extension.properties")) ||
+                                Files.exists(project.getClassesDir().resolve("META-INF/quarkus-build-steps.list"))) {
+                            continue;
                         }
-                        addProject(devModeContext, project);
-                        pomFiles.add(project.getDir().resolve("pom.xml"));
                     }
-                    repoSystem = MavenRepoInitializer.getRepositorySystem(repoSession.isOffline(), localProject.getWorkspace());
-                }
-
-                appModel = new BootstrapAppModelResolver(MavenArtifactResolver.builder()
-                        .setRepositorySystem(repoSystem)
-                        .setRepositorySystemSession(repoSession)
-                        .setRemoteRepositories(repos)
-                        .setWorkspace(localProject.getWorkspace())
-                        .build())
-                                .setDevMode(true)
-                                .resolveModel(localProject.getAppArtifact());
-                if (appModel.getAllDependencies().isEmpty()) {
-                    throw new RuntimeException("Unable to resolve application dependencies");
+                    addProject(devModeContext, project);
+                    pomFiles.add(project.getDir().resolve("pom.xml"));
                 }
-            } catch (Exception e) {
-                throw new MojoExecutionException("Failed to resolve Quarkus application model", e);
+                repoSystem = MavenRepoInitializer.getRepositorySystem(repoSession.isOffline(), localProject.getWorkspace());
             }
-            for (AppDependency appDep : appModel.getAllDependencies()) {
-                addToClassPaths(classPathManifest, devModeContext, appDep.getArtifact().getPath().toFile());
+            DefaultArtifact bootstrap = new DefaultArtifact("io.quarkus", "quarkus-development-mode", "jar",
+                    pluginDef.getVersion());
+            MavenArtifactResolver mavenArtifactResolver = MavenArtifactResolver.builder()
+                    .setRepositorySystem(repoSystem)
+                    .setRepositorySystemSession(repoSession)
+                    .setRemoteRepositories(repos)
```

Comment:
I think `.setWorkspace(localProject.getWorkspace())` should still be necessary. Otherwise, it won't be able to resolve artifacts that are supposed to be produced by the workspace but haven't actually been installed.

Reply:
Actually, you don't need to set the workspace here since you are not resolving any app deps with this resolver. Which means you can also simply use `DevMojo.this.repoSystem` and remove `repoSystem` local variable.